From: Randy Baumgarte <randy@fbn.cx>
Date: Tue, 3 Jan 2017 22:05:08 -0500
Subject: Fix issue where tags could not be deleted from a note under Qt5.

---
 sql/notetable.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sql/notetable.cpp b/sql/notetable.cpp
index a3126ed..db3c5f6 100644
--- a/sql/notetable.cpp
+++ b/sql/notetable.cpp
@@ -1326,8 +1326,8 @@ void NoteTable::removeTag(qint32 lid, qint32 tag, bool isDirty = false) {
     query.prepare("delete from DataStore where lid=:lid and key=:key and data=:tag");
     query.bindValue(":lid", lid);
     query.bindValue(":key",NOTE_TAG_LID);
-    query.bindValue(":tag:", tag);
-    query.exec();
+    query.bindValue(":tag", tag);
+    query.exec();;
     query.finish();
     db->unlock();
     if (isDirty) {
